lte/cloud/go/tools/migrations/m013_policy_ipv6/main.go

```diff
@@ -0,0 +1,233 @@
+
+import (
+	"database/sql"
+	"encoding/json"
+	"flag"
+
+	"github.com/Masterminds/squirrel"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/golang/glog"
+	_ "github.com/lib/pq"
+	"github.com/pkg/errors"
+
+	"magma/lte/cloud/go/tools/migrations/m013_policy_ipv6/types"
+	"magma/orc8r/cloud/go/sqorc"
+	"magma/orc8r/cloud/go/tools/migrations"
+)
+
+// Duplicated from configurator
+const (
+	entityTable      = "cfg_entities"
+	entityAssocTable = "cfg_assocs"
+
+	entConfCol = "config"
+	entGidCol  = "graph_id"
+	entKeyCol  = "\"key\""
+	entNidCol  = "network_id"
+	entPkCol   = "pk"
+	entTypeCol = "type"
+)
+
+const (
+	basenameEntType   = "base_name"
+	policyEntType     = "policy"
+	subscriberEntType = "subscriber"
+	qosProfileEntType = "policy_qos_profile"
+)
+
+var (
+	dryRun bool
+)
+
+type policyEnt struct {
+	pk      string
+	network string
+	gid     string
+	config  types.OldPolicyRuleConfig
+}
+
+func init() {
+	flag.BoolVar(&dryRun, "dry", false, "don't commit changes")
+	flag.Parse()
+	_ = flag.Set("alsologtostderr", "true") // enable printing to console
+}
+
+// main runs the policy_qos migration.
+// Migration without SQL error will result in `SUCCESS` printed to stderr.
+func main() {
+	defer glog.Flush()
+
+	glog.Info("BEGIN MIGRATION")
+
+	dbDriver := migrations.GetEnvWithDefault("SQL_DRIVER", "postgres")
+	dbSource := migrations.GetEnvWithDefault("DATABASE_SOURCE", "dbname=magma_dev user=magma_dev password=magma_dev host=postgres sslmode=disable")
+	db, err := sql.Open(dbDriver, dbSource)
+	if err != nil {
+		glog.Fatal(errors.Wrap(err, "could not open db connection"))
+	}
+
+	_, err = migrations.ExecInTx(db, &sql.TxOptions{Isolation: sql.LevelSerializable}, nil, doMigration)
+	if err != nil {
+		glog.Fatalf("%+v", err)
+	}
+
+	glog.Info("SUCCESS")
+	glog.Info("END MIGRATION")
+}
+
+func doMigration(tx *sql.Tx) (interface{}, error) {
+	sc := squirrel.NewStmtCache(tx)
+	defer func() { _ = sc.Clear() }()
+	builder := sqorc.GetSqlBuilder().RunWith(sc)
+
+	edges := []migrations.AssocDirection{
+		{FromType: subscriberEntType, ToType: basenameEntType}, // subscriber -> base_name
+		{FromType: subscriberEntType, ToType: policyEntType},   // subscriber -> policy
+		{FromType: basenameEntType, ToType: policyEntType},     // base_name -> policy
+	}
+
+	err := migrations.SetAssocDirections(builder, edges)
+	if err != nil {
+		return nil, err
+	}
+
+	err = migratePolicyRules(tx, builder)
+	if err != nil {
+		return nil, err
+	}
+
+	if dryRun {
+		return nil, errors.New("throwing error instead of committing because -dry was specified")
+	}
+
+	return nil, nil
+}
+
+// migratePolicyRules derives and creates a policy_qos_profile entity for each
+// existing policy entity.
+// Also removes the Qos field from existing policy entities.
+func migratePolicyRules(tx *sql.Tx, builder squirrel.StatementBuilderType) error {
+	// Get all existing policy ents
+	rows, err := builder.
+		Select(entPkCol, entKeyCol, entNidCol, entGidCol, entConfCol).
```

Comment:
you don't need all these fields I think all you need is the pk and the config. maybe the key would be useful for error logging